src/builders.ts

```diff
@@ -183,40 +232,43 @@ export class BomBuilder {
   private async makeComponentFromPackage (
     pkg: Package,
     fetchManifest: ManifestFetcher,
+    fetchLicenseEvidence: LicenseEvidenceFetcher,
     type?: ComponentType | undefined
   ): Promise<Component | false | undefined> {
-    const data = await fetchManifest(pkg)
+    const manifest = await fetchManifest(pkg)
     // the data in the manifest might be incomplete, so lets set the properties that yarn discovered and fixed
     /* eslint-disable-next-line @typescript-eslint/strict-boolean-expressions */
-    data.manifest.name = pkg.scope ? `@${pkg.scope}/${pkg.name}` : pkg.name
-    data.manifest.version = pkg.version
-    return this.makeComponent(pkg, data, type)
+    manifest.name = pkg.scope ? `@${pkg.scope}/${pkg.name}` : pkg.name
+    manifest.version = pkg.version
+    const component = this.makeComponent(pkg, manifest, type)
+    if (component instanceof Component) {
```

Comment:
I don't see how `makeComponent` could conditionally return an instance. I'd guess something is wrong if we do not get an `Component` instance here and thus the code should abort execution.

Reply:
lets discuss improvements in a different issue/pullrequest (see https://github.com/AugustusKling/cyclonedx-node-yarn/pull/1#discussion_r1841854471)